### PR TITLE
Add appSecurity-3.0, batch-1.0, and ejb-3.2 to EE8 convenience features

### PR DESCRIPTION
**Pull request by aguibert**

### CODE REVIEW

**Comment by aguibert** [comment]
#build

**Comment by aguibert** [comment]
the javax.base feature name is a long story, but it's actually a pretty clever mechanism that Rick came up with to allow both JPA 2.0 and JPA 2.1 API to be available on the server without conflicting when the `jpa-2.0` and `persistence-1.0` features are enabled.  

Note that it's an internal feature, so end users will never see this feature being enabled in server output

but anyway, that is just extending an existing mechanism for JPA 2.2.  Last week Joe and I put in some feature refactoring to allow the `persistence-1.0` internal feature to work with either JPA 2.1 or JPA 2.2

**Comment by aguibert** [comment]
regarding the `javax.enterprise` stuff, this was needed to allow the Batch + CDI auto feature to work with CDI 2.0, so I had to expand the version range on the import

File: dev/com.ibm.websphere.appserver.javaee-8.0/com.ibm.websphere.appserver.javaee-8.0.feature [hunk 1]
```diff
@@ -5,14 +5,17 @@ IBM-App-ForceRestart: install, uninstall
...
 Subsystem-Name: Java EE Full Platform 8.0
-# TODO-EE8 add batch-1.0 and ejb-3.2 once persistence service works w/ JPA 2.2
 -features=\
   com.ibm.websphere.appserver.appClientSupport-1.0,\
+  com.ibm.websphere.appserver.batch-1.0,\
   com.ibm.websphere.appserver.concurrent-1.0,\
+  com.ibm.websphere.appserver.ejb-3.2,\
   com.ibm.websphere.appserver.jacc-1.5,\
   com.ibm.websphere.appserver.javaMail-1.6,\
+  com.ibm.websphere.appserver.javax.persistence.base-2.2,\
```
[the remark this answers is not in the document]
Comment (aguibert) [reply dev/com.ibm.websphere.appserver.javaee-8.0/com.ibm.websphere.appserver.javaee-8.0.feature]: the javax.base feature name is a long story, but it's actually a pretty clever mechanism that Rick came up with to allow both JPA 2.0 and JPA 2.1 API to be available on the server without conflicting when the jpa-2.0 and persistence-1.0 features are enabled.

Note that it's an internal feature, so end users will never see this feature being enabled in server output

but anyway, that is just extending an existing mechanism for JPA 2.2. Last week Joe and I put in some feature refactoring to allow the persistence-1.0 internal feature to work with either JPA 2.1 or JPA 2.2

[the remark this answers is not in the document]
Comment (kwsutter) [reply dev/com.ibm.websphere.appserver.javaee-8.0/com.ibm.websphere.appserver.javaee-8.0.feature]: Thanks for the clarification, Andy.  As long as javax.persistence.base-2.2 is private, then I'm good.  On first glance, I thought it was a public feature and I didn't want that feature name to be usable by customers.

Via a side discussion with @aguibert, I also learned that this particular feature has to be specified in the javaee-8.0 feature to provide for feature toleration down the chain.  (I was asking why this feature couldn't just be specified in batch-1.0 and ejb-3.2 features.)

**Pull request merged**